src/gc/gc.d

```diff
@@ -3132,19 +3134,17 @@ struct SmallObjectPool
 
                 rt_finalizeFromGC(q, size, attr);
 
-                toClear |= GCBits.BITS_1 << clearIndex;
+                freeBits = true;
+                toFree.set(i);
 
                 debug(COLLECT_PRINTF) printf("\tcollecting %p\n", p);
                 //log_free(sentinel_add(p));
 
                 debug (MEMSTOMP) memset(p, 0xF3, size);
-                freebits.set(biti);
```

Comment:
Neither the function name nor the API description make it obvious that the objects are actually freed. If this is desired, the object should be added to the bucket list instead of setting freebits. BTW: there are some new unitests in lifetime that manually free objects after runFinalizers.